src/nfa/nfa-concatenation.cc

```diff
@@ -26,6 +26,43 @@ Nfa concatenate(const Nfa& lhs, const Nfa& rhs, bool use_epsilon,
     return Algorithms::concatenate_eps(lhs, rhs, EPSILON, use_epsilon, lhs_result_states_map, rhs_result_states_map);
 }
 
+Nfa& Nfa::concatenate(const Nfa& aut) {
+    size_t n = this->size();
+    auto upd_fnc = [&](State st) {
+        return st + n;
+    };
+
+    this->delta.append(aut.delta.transform(upd_fnc));
+
+    // set accepting states
+    Util::SparseSet<State> new_fin{};
+    new_fin.reserve(n+aut.size());
+    for(const State& aut_fin : aut.final) {
+        new_fin.insert(upd_fnc(aut_fin));
+    }
```

Comment:
should we do it using SparseSet.rename()?
Currently not sure that it would work, since it was made assuming that renaming gives a smaller name, where here it gives a larger name. Myabe we would need two variants or to fix it somehowe, but anyway, should we do such things using methods like rename?

Reply:
Probably yes, in general. In this simple case I am not sure. But I am pretty sure that it won't be more efficient (you need to copy the original SparseSet and then to rename). Here it is a single loop only.